Package: centralize description string handling

cassidyjames opened the pull request:
Fixes #1700

davidmhewitt commented on src/Widgets/Banner.vala:
```diff
@@ -57,11 +57,14 @@ public class AppCenter.Widgets.Banner : Gtk.Button {
+            }
         }
 
         int close_paragraph_index = description.index_of ("</p>", 0);
```
Not directly related to your PR, which seems logical, but I'm not a fan of this `index_of` a thing that might not be there, and the subsequent hardcoded `slice` either.

I think I'd like to have this PR as an opportunity to at least comment why half of this stuff is happening or fix it up

cassidyjames commented:
@davidmhewitt it looks like `AppStream.markup_convert_simple ()` does a lot of what we were trying to do manually, so I simplified a bit to use that and added a comment about grabbing only the first line for the banner. Hopefully this makes more sense?

cassidyjames commented:
Actually, this doesn't quite do all the same things; if someone has

```html
<p>
    This is line             one of the description!
    And this is another line but should be part of the same paragraph.
</p>
```

This doesn't strip the internal whitespace (like HTML would), and it still treats the two lines as separate lines. Since we always want to follow the same cleaned-up formatting rules here, I think it makes more sense to implement whatever transforming of the string we want directly in `Package.get_description ()`, and then we can more clearly comment and step through each part so it's much simpler to just grab and display that description wherever we want.

If that approach makes more sense, I'll pursue that instead.

davidmhewitt commented on src/Core/Package.vala:
```diff
@@ -626,13 +625,28 @@ public class AppCenterCore.Package : Object {
 
                 description = backend_details.description;
             }
+
```
Add a `description == null` check here and return `null` if so. It's possible that a package still won't have a description even after trying two different ways to get it.

davidmhewitt replied:
Worth noting that this change would resolve the segfault that @danrabbit was seeing when opening the "OS Updates" component while AppCenter was still loading.

danrabbit reviewed (approved):
@davidmhewitt do you still have changes you'd like to see here?

I can confirm that the crash I was experiencing has been resolved
